API/Repositories/Implementations/MatchHistoryRepository.cs

```diff
@@ -0,0 +1,26 @@
+﻿using API.Entities;
+using API.Enums;
+using API.Repositories.Interfaces;
+using AutoMapper;
+
+namespace API.Repositories.Implementations;
+
+public class MatchHistoryRepository : RepositoryBase<MatchHistory>, IHistoryRepository<MatchHistory, Match>
+{
+    private readonly IMapper _mapper;
+
+    public MatchHistoryRepository(OtrContext context, IMapper mapper)
+        : base(context)
+    {
+        _mapper = mapper;
+    }
+
+    public async Task<MatchHistory?> CreateAsync(Match match, HistoryActionType action)
+    {
+        MatchHistory record = _mapper.Map<MatchHistory>(match);
+        record.HistoryAction = (int)action;
+        // Modifications made automatically (without user intervention) have an Id of null
+        record.ModifierId = null;
+        return await base.CreateAsync(record);
+    }
```

Comment:
I think this & some other common methods (such as updated [by user]) could go into a HistoryRepositoryBase class and be made generic. Don't want to do this for each history item if the logic is the same.